src/main/scala/io/iohk/praos/crypto/VerifiableRandomFunction.scala

```diff
@@ -1,34 +1,46 @@
 package io.iohk.praos.crypto
 
 import akka.util.ByteString
+import scala.math.abs
+
+trait VerifiableRandomFunction {
+
+  type VrfProof = ByteString
+
+  /**
+    * @return (nonce, vrfProof)
+    */
+  def prove(privateKey: Key, randomSeed: RandomValue): (RandomValue, VrfProof)
+
+  def verify(publicKey: Key, randomSeed: RandomValue, vrfProof: VrfProof): Boolean
+}
 
 /**
   * VrfProof contains a fresh random value generated using the user public key, a random seed and
   * VerifiableRandomFunction.prove method. And also contains the proof, that will be used later,
   * with the user public key, for verify that the random value was generated by him.
   */
-case class VrfProof(random: RandomValue, proof: ByteString)
 
 object VrfProof {
   // TODO: When the VrfProof be well defined, change to the true random length
   val randomLength = 32
 }
 
-trait VerifiableRandomFunction {
-
-  def prove(privateKey: Key, randomSeed: RandomValue): VrfProof
-
-  def verify(publicKey: Key, randomSeed: RandomValue, vrfProof: VrfProof): Boolean
-}
-
 /**
   * This is a fake implementation that fullfills VerifiableRandomFunction properties
   */
 object VerifiableRandomFunctionStubImpl extends VerifiableRandomFunction {
 
-  def prove(privateKey: Key, randomSeed: RandomValue): VrfProof =
-    VrfProof(randomSeed, getPublicKeyFromPrivateKey(privateKey))
+  private val hasher = PredefinedHasher("MD5")
+
+  def prove(privateKey: Key, randomSeed: RandomValue): (RandomValue, VrfProof) = {
+    val publicKey = getPublicKeyFromPrivateKey(privateKey)
+    (
+      abs(BigInt(hasher.hash(publicKey ++ ByteString(randomSeed)).toArray).toInt),
+      publicKey
+    )
+  }
 
   def verify(publicKey: Key, randomSeed: RandomValue, vrfProof: VrfProof): Boolean =
```

Comment:
Should the `RandomValue` returned by `prove` be also a parameter of this function? (as in [here](https://github.com/atixlabs/praos_test/pull/11/files#diff-ddaeabc85998b62b41e40db26f082065R12)). Should we also verify it was correctly obtained as well?